### PR TITLE
fix(platform-server): add styles to elements correctly

### LINKED ISSUE #22536: bug(platform-server): some styles not applying correctly
After #22263, some styles like `background-image` are not applied correctly by Domino. This is because the style map is no longer getting written each time, and Domino's internal representation is not implemented well enough to accommodate this.

### PR DESCRIPTION
**Pull request by CaerusKaru**
* Partially reverts #22263 due to lack of total spec complicance
  on the server
* Maintains the camel-case styles fix

Fixes #22536

### CODE REVIEW

**Comment by CaerusKaru** [comment]
@vikerman @alexeagle

**Comment by vicb** [comment]
1) What was the issue ? (could you please create a GH issue and link it from the test)
2) could you please rephrase the commit message:
```
fix(platform-server): <what does this fix ?>

partial revert of #22263
```

File: packages/platform-server/src/domino_adapter.ts [hunk 1]
```diff
@@ -135,17 +135,50 @@ export class DominoAdapter extends BrowserDomAdapter {
     return href;
   }
 
+  /** @internal */
+  _readStyleAttribute(element: any) {
+    const styleMap = {};
+    const styleAttribute = element.getAttribute('style');
+    if (styleAttribute) {
+      const styleList = styleAttribute.split(/;+/g);
```
Comment (vicb) [inline packages/platform-server/src/domino_adapter.ts]: nit: what about `/[\s;]+/g` here ?

Comment (vicb) [reply packages/platform-server/src/domino_adapter.ts]: oh would not work either so maybe 
1- split on `/;+/g`
2- `trim()` the parts
3- ignore empty parts

Comment (CaerusKaru) [reply packages/platform-server/src/domino_adapter.ts]: I changed the logic, let me know if that's what you had in mind

**Comment by CaerusKaru** [comment]
@vicb changed commit message and opened a new issue

**Comment by agustinhaller** [comment]
@vicb any updates on merging this PR?

File: packages/platform-server/src/domino_adapter.ts [hunk 2]
```diff
@@ -135,17 +135,53 @@ export class DominoAdapter extends BrowserDomAdapter {
...
+      for (let i = 0; i < styleList.length; i++) {
+        if (styleList[i].length > 0) {
```
Comment (vicb) [inline packages/platform-server/src/domino_adapter.ts]: `trim`ing before checking the length would probably make more sense ?

File: packages/platform-server/src/domino_adapter.ts [hunk 3]
```diff
@@ -135,17 +135,53 @@ export class DominoAdapter extends BrowserDomAdapter {
...
+          const style = (styleList[i] as string).trim();
+          if (!style) {
+            continue;
+          }
+          const colon = style.indexOf(':');
```
Comment (vicb) [inline packages/platform-server/src/domino_adapter.ts]: nit colon -> colonIndex

File: packages/platform-server/src/domino_adapter.ts [hunk 4]
```diff
@@ -135,17 +135,53 @@ export class DominoAdapter extends BrowserDomAdapter {
...
+          if (colon === -1) {
+            throw new Error(`Invalid CSS style: ${style}`);
+          }
+          (styleMap as any)[style.substr(0, colon).trim()] = style.substr(colon + 1).trim();
```
Comment (vicb) [inline packages/platform-server/src/domino_adapter.ts]: nit: extract `const name = style.substr(0, colon).trim();`

File: packages/platform-server/src/domino_adapter.ts [hunk 5]
```diff
@@ -135,17 +135,53 @@ export class DominoAdapter extends BrowserDomAdapter {
     return href;
   }
 
+  /** @internal */
+  _readStyleAttribute(element: any) {
+    const styleMap = {};
```
Comment (vicb) [inline packages/platform-server/src/domino_adapter.ts]: `const styleMap: {[name: string]: string} = {}` and then remove un-needed "as any"

File: packages/platform-server/src/domino_adapter.ts [hunk 6]
```diff
@@ -135,17 +135,53 @@ export class DominoAdapter extends BrowserDomAdapter {
     return href;
   }
 
+  /** @internal */
+  _readStyleAttribute(element: any) {
```
Comment (vicb) [inline packages/platform-server/src/domino_adapter.ts]: add return type

File: packages/platform-server/src/domino_adapter.ts (hunk 4 above)
Comment (vicb) [reply packages/platform-server/src/domino_adapter.ts]: there should be `as any` needed here

File: packages/platform-server/src/domino_adapter.ts [hunk 7]
```diff
@@ -135,17 +135,53 @@ export class DominoAdapter extends BrowserDomAdapter {
...
+        styleAttrValue += key + ':' + styleMap[key] + ';';
+      }
+    }
+    element.setAttribute('style', styleAttrValue);
+  }
   setStyle(element: any, styleName: string, styleValue?: string|null) {
     styleName = styleName.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase();
-    element.style[styleName] = styleValue;
+    const styleMap = this._readStyleAttribute(element);
+    (styleMap as any)[styleName] = styleValue;
```
Comment (vicb) [inline packages/platform-server/src/domino_adapter.ts]: no need for any if types are specified upstream

File: packages/platform-server/src/domino_adapter.ts [hunk 8]
```diff
@@ -135,17 +135,53 @@ export class DominoAdapter extends BrowserDomAdapter {
...
   removeStyle(element: any, styleName: string) {
     // IE requires '' instead of null
     // see https://github.com/angular/angular/issues/7916
-    element.style[styleName] = '';
+    this.setStyle(element, styleName, '');
   }
   getStyle(element: any, styleName: string): string {
-    return element.style[styleName] || element.style.getPropertyValue(styleName);
+    const styleMap = this._readStyleAttribute(element);
+    return styleMap.hasOwnProperty(styleName) ? (styleMap as any)[styleName] : '';
```
Comment (vicb) [inline packages/platform-server/src/domino_adapter.ts]: no need for as any

**Review by vicb — CHANGES_REQUESTED** [review]
a few nits but LGTM otherwise.
Thanks  and sorry for the delay in reviewing

**Comment by vicb** [comment]
- Typo in the commit message (complicance)
- Was is the target: master and/or patch

File: packages/platform-server/src/domino_adapter.ts (hunk 5 above)
Comment (CaerusKaru) [reply packages/platform-server/src/domino_adapter.ts]: Fixed

File: packages/platform-server/src/domino_adapter.ts (hunk 4 above)
Comment (CaerusKaru) [reply packages/platform-server/src/domino_adapter.ts]: Fixed

File: packages/platform-server/src/domino_adapter.ts (hunk 3 above)
Comment (CaerusKaru) [reply packages/platform-server/src/domino_adapter.ts]: Fixed

File: packages/platform-server/src/domino_adapter.ts (hunk 2 above)
Comment (CaerusKaru) [reply packages/platform-server/src/domino_adapter.ts]: Fixed

File: packages/platform-server/src/domino_adapter.ts (hunk 6 above)
Comment (CaerusKaru) [reply packages/platform-server/src/domino_adapter.ts]: Fixed

File: packages/platform-server/src/domino_adapter.ts (hunk 7 above)
Comment (CaerusKaru) [reply packages/platform-server/src/domino_adapter.ts]: Fixed

File: packages/platform-server/src/domino_adapter.ts (hunk 8 above)
Comment (CaerusKaru) [reply packages/platform-server/src/domino_adapter.ts]: Fixed

**Comment by CaerusKaru** [comment]
Should be `PR target: master & patch`

File: packages/platform-server/src/domino_adapter.ts [hunk 9]
```diff
@@ -135,17 +135,51 @@ export class DominoAdapter extends BrowserDomAdapter {
...
+        styleAttrValue += key + ':' + styleMap[key] + ';';
+      }
+    }
+    element.setAttribute('style', styleAttrValue);
+  }
   setStyle(element: any, styleName: string, styleValue?: string|null) {
     styleName = styleName.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase();
-    element.style[styleName] = styleValue;
+    const styleMap = this._readStyleAttribute(element);
+    styleMap[styleName] = styleValue + '';
```
Comment (vicb) [inline packages/platform-server/src/domino_adapter.ts]: do we really want `"undefined"` as a possible value ?

File: packages/platform-server/src/domino_adapter.ts [hunk 10]
```diff
@@ -135,17 +135,51 @@ export class DominoAdapter extends BrowserDomAdapter {
...
   removeStyle(element: any, styleName: string) {
     // IE requires '' instead of null
     // see https://github.com/angular/angular/issues/7916
-    element.style[styleName] = '';
+    this.setStyle(element, styleName, '');
   }
   getStyle(element: any, styleName: string): string {
-    return element.style[styleName] || element.style.getPropertyValue(styleName);
+    const styleMap = this._readStyleAttribute(element);
+    return styleMap.hasOwnProperty(styleName) ? styleMap[styleName] : '';
```
Comment (vicb) [inline packages/platform-server/src/domino_adapter.ts]: `return styleMap[styleName] || '';` ?

File: packages/platform-server/src/domino_adapter.ts (hunk 9 above)
Comment (CaerusKaru) [reply packages/platform-server/src/domino_adapter.ts]: Fixed

File: packages/platform-server/src/domino_adapter.ts (hunk 10 above)
Comment (CaerusKaru) [reply packages/platform-server/src/domino_adapter.ts]: Fixed

**Review by vicb — APPROVED** [review]
Thanks for the updates !

**Comment by vicb** [comment]
g3 status: tap/189074679 
Thanks @kara

**Review by vikerman — APPROVED** [review]

**Pull request closed without merging**